include/seqan3/alignment/pairwise/alignment_algorithm.hpp

```diff
@@ -644,40 +643,62 @@ class alignment_algorithm :
+     *
+     * If the alignment is run in debug mode (see seqan3::align_cfg::debug) the debug score and optionally trace matrix
+     * are stored in the alignment result as well.
+     */
+    template <typename indexed_sequence_pair_range_t>
     //!\cond
         requires traits_t::is_vectorised
     //!\endcond
-    constexpr auto make_alignment_result(index_range_t && index_range,
-                                         sequence1_range_t && sequence1_range,
-                                         [[maybe_unused]] sequence2_range_t && sequence2_range)
+    constexpr auto make_alignment_result(indexed_sequence_pair_range_t && index_sequence_pairs)
     {
-        using sequence1_t = std::ranges::range_value_t<sequence1_range_t>;
-        using sequence2_t = std::ranges::range_value_t<sequence2_range_t>;
+        using indexed_sequence_pair_t = std::ranges::range_value_t<indexed_sequence_pair_range_t>;
+        using sequence_pair_t = std::tuple_element_t<0, indexed_sequence_pair_t>;
+        using sequence1_t = std::tuple_element_t<0, sequence_pair_t>;
+        using sequence2_t = std::tuple_element_t<1, sequence_pair_t>;
 
         using result_value_t = typename align_result_selector<sequence1_t, sequence2_t, config_t>::type;
 
-        size_t number_of_computed_alignments = std::ranges::distance(sequence1_range);
-        auto index_it = std::ranges::begin(index_range);
-        std::vector<alignment_result<result_value_t>> results{number_of_computed_alignments};
-        // Note we are not using the index range as it might be an infinity range.
-        for (size_t i = 0; i < number_of_computed_alignments; ++i, ++index_it)
+        std::vector<alignment_result<result_value_t>> results{};
+        results.reserve(std::ranges::distance(index_sequence_pairs));
+
+        size_t simd_index = 0;
+        for (auto && [sequence_pairs, alignment_index] : index_sequence_pairs)
         {
+            (void) sequence_pairs;
             result_value_t res{};
-            res.id = *index_it;
+            res.id = alignment_index;
 
-            // Choose what needs to be computed.
-            if constexpr (traits_t::result_type_rank >= 0)  // compute score
-                res.score = this->alignment_state.optimum.score[i];
+            res.score = this->alignment_state.optimum.score[simd_index];  // Just take this
 
             if constexpr (traits_t::result_type_rank >= 1)  // compute back coordinate
             {
-                res.back_coordinate.first = this->alignment_state.optimum.column_index[i];
-                res.back_coordinate.second = this->alignment_state.optimum.row_index[i];
+                res.back_coordinate.first = this->alignment_state.optimum.column_index[simd_index] ;
+                res.back_coordinate.second = this->alignment_state.optimum.row_index[simd_index];
             }
 
-            results[i] = std::move(res);
+            results.emplace_back(std::move(res));
+            ++simd_index;
```

Comment:
In the documentation you state that the fron_coordinate, alignment and debug information are also computed if requested but I don't see this reflected in the code?

Reply:
This will follow in the next steps.